File: app/src/main/java/com/hippo/ehviewer/dao/QuickSearchDao.java
package com.hippo.ehviewer.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface QuickSearchDao {

    String TABLENAME = "QUICK_SEARCH";

    @Query("SELECT * FROM QUICK_SEARCH ORDER BY TIME ASC")
    List<QuickSearch> list();

    @Query("SELECT * FROM QUICK_SEARCH ORDER BY TIME ASC LIMIT :limit OFFSET :offset")
    List<QuickSearch> list(int offset, int limit);

    @Update
    void update(List<QuickSearch> downloadLabels);

    @Update
    void update(QuickSearch quickSearch);

    @Insert
    long insert(QuickSearch quickSearch);

    @Delete
    void delete(QuickSearch quickSearch);

}

File: app/src/main/java/com/hippo/ehviewer/dao/QuickSearchDao.kt
package com.hippo.ehviewer.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface QuickSearchDao {
    @Query("SELECT * FROM QUICK_SEARCH ORDER BY POSITION ASC")
    suspend fun list(): List<QuickSearch>

    @Query("SELECT * FROM QUICK_SEARCH ORDER BY POSITION ASC LIMIT :limit OFFSET :offset")
    suspend fun list(offset: Int, limit: Int): List<QuickSearch>

    @Query("UPDATE QUICK_SEARCH SET POSITION = POSITION - 1 WHERE POSITION > :position")
    suspend fun fill(position: Int)

    @Update
    suspend fun update(quickSearchList: List<QuickSearch>)

    @Insert
    suspend fun insert(quickSearch: QuickSearch): Long

    @Insert
    suspend fun insert(quickSearchList: List<QuickSearch>)

    @Delete
    suspend fun delete(quickSearch: QuickSearch)
}
